### DISCUSSION TITLE
Will I end up with two versions of Zod now that Next includes it?

### DISCUSSION THREAD

**Discussion by philwolstenholme** [vercel/next.js#48079] [Help]
### Summary

I noticed that Next now includes Zod as a precompiled dependency: https://github.com/vercel/next.js/pull/46962

Given that the dependency isn't managed through npm, will this lead to projects shipping Zod to their users twice if Next includes it in its own bundles, and my project already includes it via npm for our own application code? Or is Webpack smart enough to deduplicate?

If there is a risk of duplication, then is it possible for me to import the Next precompiled version and remove Zod from my package.json? Has anyone done a similar thing before for the Next precompiled dependencies?

**Comment by ivan-kleshnin** [comment]
[@philwolstenholme](https://github.com/philwolstenholme) your intuition is on point.

This already hit me. Got two Zod versions within a monorepo. Getting weird bugs that are impossible to reproduce outside the project scope 😨 

```
node_modules
package.json <-- defines a common zod version for the project (v22)
next-app
  package.json <-- no "zod" here 
  yarn.lock <-- mentions another, incompatible zod version from NextJS (v21)
```

Result:
```
ZodReadonly (something from Zod v22) is not found
```

within `next-app` folder for some (but not all) type assigments. Zod should have been added as a peer dependency for NextJS.
